### DISCUSSION TITLE
Auto update language packs

### DISCUSSION THREAD

**Discussion by MCHuijgen** [mozilla/policy-templates#952] [General]
Hi Mike, 

At my organization, we're using the en-US and fy-NL language pack for our nl-NL Firefox. I'm using the policy ExtensionSettings for deploying them (see below), but I have to update the GPO with every major version of Firefox. If we don't, with fresh deployments of Firefox, the language packs won't be active because they're not compatible, and the auto extension update feature won't update the extension straight away at the first start of Firefox.
```
      "langpack-fy-NL@firefox.mozilla.org": {
        "installation_mode": "force_installed",
        "install_url": "https://releases.mozilla.org/pub/firefox/releases/105.0/win64/xpi/fy-NL.xpi"
      },
      "langpack-en-US@firefox.mozilla.org": {
        "installation_mode": "force_installed",
        "install_url": "https://releases.mozilla.org/pub/firefox/releases/105.0/win64/xpi/en-US.xpi"
      }
```
On https://support.mozilla.org/nl/questions/1285159 I saw your suggestion to put the language files in the extension folder, but by that, I still have to update it with every upgrade. Am I doing this in a wrong way, or would you recommend that I create a feature request in Bugzilla to create a same way for updating extensions with 'latest.xpi'? Like this:
```
     "fy@dictionaries.addons.mozilla.org": {
        "installation_mode": "normal_installed",
        "install_url": "https://addons.mozilla.org/firefox/downloads/latest/dictionnaire-frison/latest.xpi"
      },
```
This would really help for creating a 'zero touch' deployment in our organization.

**Comment by mkaply** [comment]
We put in a great deal of work to make sure language packs get updates when Firefox gets updated, so I'm surprised you're seeing any update issues.

Or is this primarily about setting up new deployments?

**Reply by MCHuijgen** [reply]
Indeed, it's about new deployments. With new deployments, when I define the install_url as not the same major version as Firefox, all other extensions get installed almost instantly, except for the language packs. I'm seeing them as available updates (see attached image), but they won't be installed (I also don't see the xpi files added to the extensions folder in the profile folder). Not even after restarts of Firefox or when keeping Firefox active for a period of time. Only when I match the install_url with the major version of Firefox, the language packs get installed right away.
![Firefox Language Pack not updated](https://user-images.githubusercontent.com/49674916/193015767-2e6aa3c5-77e7-41c5-943e-cd115c2cb78e.png)

**Reply by MCHuijgen** [reply]
Hi Mike, can you reproduce this? And if so, can something be done to have the language pack updates installed immediately?

**Comment by mkaply** [comment]
I've opened a bug to see if we can get URLs to the release language packs.

https://bugzilla.mozilla.org/show_bug.cgi?id=1799684

That would be the thing that would help here.

**Reply by MCHuijgen** [reply]
Great, thanks for your effort!

**Reply by MCHuijgen** [reply]
Hi Mike, any updates on this?

**Comment by mkaply** [comment]
No updates. I'll ping in the bug.

**Comment by mkaply** [comment]
So we now have this.

A URL like

https://download.mozilla.org/?product=firefox-langpack-latest-SSL&lang=fr

will get you the latest language pack for french.

I need to do some better docs. but we have something

**Reply by MCHuijgen** [reply]
Thanks so much, works like a charm!
